BUG: `TestNNLS.test_nnls_inner_loop_case1` fails with MKL

This already showed up in the 1.13 [release PR](https://github.com/conda-forge/scipy-feedstock/pull/271) on the conda-forge feedstock, but was deferred at the time. Further [testing](https://github.com/conda-forge/scipy-feedstock/pull/273) showed that it only happens with the MKL-flavour of BLAS/LAPACK, and even more particularly, only in the presence of some AVX512 instructions:

```
_____________________ TestNNLS.test_nnls_inner_loop_case1 _____________________
[gw0] win32 -- Python 3.9.19 %PREFIX%\python.exe
..\_test_env\lib\site-packages\scipy\optimize\tests\test_nnls.py:101: in test_nnls_inner_loop_case1
    dact, _ = nnls(W @ A, W @ k)
        2.23606798,  2.23606798,  1.        ,  1.41421356,  1.73205081,
        1.        ,  1.41421356])
..\_test_env\lib\site-packages\scipy\optimize\_nnls.py:93: in nnls
    raise RuntimeError("Maximum number of iterations reached.")
E   RuntimeError: Maximum number of iterations reached.
        A          = array([[ 5.19615242,  0.        ,  0.        , ...,  0.        ,
         0.        ,  0.        ],
       [ 2.8284271...   , -1.        ],
       [ 2.82842712,  2.82842712,  2.82842712, ...,  2.82842712,
         2.82842712,  2.82842712]])
        atol       = None
        b          = array([   0.        ,    0.        ,    0.        ,    0.        ,
          0.        ,    0.        ,    0.        ,...         11.27887984,    8.05634274,    0.        ,    4.07622283,
          0.        ,    0.        ,    0.        ])
        m          = 67
        maxiter    = None
        mode       = -1
        n          = 67
        rnorm      = 0.0
        x          = array([0.        , 0.        , 0.        , 0.        , 0.        ,
       0.        , 0.        , 0.        , 0.      ...       , 0.        ,
       0.        , 0.        , 0.        , 0.        , 0.        ,
       0.        , 0.        ])

FAILED optimize/tests/test_nnls.py::TestNNLS::test_nnls_inner_loop_case1 - RuntimeError: Maximum number of iterations reached.
```

--- a/scipy/optimize/tests/test_nnls.py
+++ b/scipy/optimize/tests/test_nnls.py
@@ -97,7 +97,7 @@ class TestNNLS:
 
         # Small perturbations can already make the infinite loop go away (just
         # uncomment the next line)
-        # k = k + 1e-10 * np.random.normal(size=N)
+        k = k + 1e-10 * np.random.normal(size=N)
         dact, _ = nnls(W @ A, W @ k)
         assert_allclose(dact, d, rtol=0., atol=1e-10)
 

